### PR TITLE
rdkafka: use openssl as dependency

### PR DESCRIPTION
**Pull request by neheb**
Should fix MSVC compilation.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

**Comment by neheb** [comment]
```
2022-11-27T03:00:55.7754914Z rdkafka| Run-time dependency libssl (modules: OpenSSL::SSL) found: YES 1.1.1i
2022-11-27T03:00:55.7755652Z rdkafka| Run-time dependency libcrypto (modules: OpenSSL::Crypto) found: YES 1.1.1i
```

Wonder where this is coming from. CI update?

**Comment by neheb** [comment]
```
2022-11-27T03:01:26.8564568Z FAILED: subprojects/librdkafka-1.9.2/rdkafka.dll subprojects/librdkafka-1.9.2/rdkafka.pdb 
2022-11-27T03:01:26.8565585Z "link" @subprojects/librdkafka-1.9.2/rdkafka.dll.rsp
2022-11-27T03:01:26.8572060Z LINK : fatal error LNK1104: cannot open file 'libcrypto.lib'
```

Interesting...

File: subprojects/packagefiles/rdkafka/meson.build [hunk 1]
```diff
@@ -103,9 +103,8 @@ curl = dependency('libcurl', required: get_option('WITH_CURL'))
 zlib = dependency('zlib', required: get_option('WITH_ZLIB'))
 zstd = dependency('libzstd', required: get_option('WITH_ZSTD'))
 lz4 = dependency('liblz4', required: get_option('WITH_LZ4_EXT'))
-sasl = dependency('libsasl2', required: get_option('WITH_SASL')) # TODO: make SASL usable without libsasl2 on windows 
-ssl = dependency('libssl', required: get_option('WITH_SSL'))
-crypto = dependency('libcrypto', required: get_option('WITH_SSL'))
+sasl = dependency('libsasl2', required: get_option('WITH_SASL')) # TODO: make SASL usable without libsasl2 on windows
+ssl = dependency('openssl', required: get_option('WITH_SSL'), method: 'pkg-config')
```
Comment (eli-schwartz) [inline subprojects/packagefiles/rdkafka/meson.build:107]: This is wrong...

The actual problem is Strawberry Perl is terrible and should be rm -rf'ed from the runner VM image, and the latest cmake trips over that broken installation.

Comment (neheb) [reply subprojects/packagefiles/rdkafka/meson.build:107]: No argument there.

**Pull request merged**